sync-core/src/main/java/com/cloudant/sync/query/QueryResult.java

```diff
@@ -55,109 +54,132 @@ public QueryResult(List<String> docIds,
     /**
      *  Returns the number of documents in this query result.
      *
-     *  @return the number of the {@code DocumentRevision} in this query result
+     *  @return the number of documents {@code DocumentRevision} in this query result.
      */
     public long size() {
-        // TODO - skip, limit w.r.t. size
-
-        return docIds.size();
+        return documentIds().size();
     }
 
     /**
      *  Returns a list of the document ids in this query result.
      *
+     *  This method is implemented this way to ensure that the list of document ids is
+     *  consistent with the iterator results.
+     *
      *  @return list of the document ids
      */
     public List<String> documentIds() {
-        // TODO - skip, limit w.r.t. document id list
-
-        return docIds;
+        List<String> documentIds = new ArrayList<String>();
+        List<BasicDocumentRevision> docs = Lists.newArrayList(iterator());
+        for (BasicDocumentRevision doc : docs) {
+            documentIds.add(doc.getId());
+        }
+        return documentIds;
     }
 
     @Override
     public Iterator<BasicDocumentRevision> iterator() {
```

Comment:
Made QueryResult implement Iterable\<DocumentRevision> and used DocumentRevision throughout the code.  The exception being the case where we are documents are retrieved from the datastore since the datastore returns List\<BasicDocumentRevision>.
https://github.com/cloudant/sync-android/commit/cf527fd540554a34338857f179c44fa358444ad3#diff-5f5fe154c6c97af9fc67a7b9f8ae9c87L29